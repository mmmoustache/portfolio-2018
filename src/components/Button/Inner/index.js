import styled from "styled-components";

export default styled.span`
  position: relative;
  z-index: 1;
`;
