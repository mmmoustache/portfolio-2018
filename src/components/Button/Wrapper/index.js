import styled from "styled-components";
import { Breakpoints } from '../../../utils/breakpoints';
import { Colours } from '../../../utils/colours';
import { Timings, Durations } from '../../../utils/transitions';
import { Border as BorderSettings } from '../../../utils/layouts';

const Small = {
  width: '200px',
};

const BorderPadding = '30px';
const Offset = '120%';
const BackgroundSkew = '-25deg';

export default styled.a`
  background: ${Colours.white};
  border: ${BorderSettings.size} solid ${Colours.black};
  color: ${Colours.black};
  display: block;
  overflow: hidden;
  padding: ${BorderPadding};
  position: relative;
  text-align: center;
  text-decoration: none;
  width: 100%;

  @media ${Breakpoints.small} {
    width: ${Small.width};
  }

  &::after,
  &::before {
    background: ${Colours.primary};
    content: '';
    display: block;
    height: 100%;
    left: -${Offset};
    position: absolute;
    top: 0;
    transform: skew(${BackgroundSkew});
    transition: ${Durations.slow} all ${Timings.default};
    width: 100%;
  }

  &::before {
    background: ${Colours.secondary};
    transition-delay: 0.1s;
  }

  &:focus,
  &:hover {
    background: ${Colours.black};
    color: ${Colours.white};

    &::after,
    &::before {
      left: ${Offset};
    }
  }
`;
