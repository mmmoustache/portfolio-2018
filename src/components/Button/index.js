import React from 'react';
import styled from "styled-components";
import H3 from "../H3";
import { Breakpoints } from '../../utils/breakpoints';

const Wrapper = styled.a`
  background: #fff;
  border: 3px solid black;
  color: black;
  display: block;
  overflow: hidden;
  padding: 30px;
  position: relative;
  text-align: center;
  text-decoration: none;
  width: 100%;

  @media ${Breakpoints.small} {
    width: 200px;
  }

  &::after,
  &::before {
    background: #F25F5C;
    content: '';
    display: block;
    height: 100%;
    width: 100%;
    left: -120%;
    position: absolute;
    top: 0;
    transform: skew(-25deg);
    transition: 0.5s all ease-in-out;
  }

  &::before {
    background: #FCA311;
    transition-delay: 0.1s;
  }

  &:focus,
  &:hover {
    background: black;
    color: white;

    &::after,
    &::before {
      left: 120%;
    }
  }
`;

const Inner = styled.span`
  position: relative;
  z-index: 1;
`;

const Button = ({ href, children }) => 
  <Wrapper href={href}><Inner><H3 noMargin>{children}</H3></Inner></Wrapper>
;

export default Button;
