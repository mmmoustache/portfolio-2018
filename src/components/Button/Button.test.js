import React from 'react';
import renderer from 'react-test-renderer';
import Button from "../Button";

describe('Component: Button', () => {

  it('Should render successfully', () => {
    const component = renderer.create(
      <Button href="/">Test</Button>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

});
