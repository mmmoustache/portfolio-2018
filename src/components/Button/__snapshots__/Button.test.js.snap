// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Button Should render successfully 1`] = `
<a
  className="sc-htpNat kReTet"
  href="/"
>
  <span
    className="sc-bwzfXH cbEDiN"
  >
    <h3
      className="sc-bdVaJa iubVGX"
    >
      Test
    </h3>
  </span>
</a>
`;
